lib/entry/FlavorDownloader.js

```diff
@@ -342,45 +344,27 @@ var FlavorDownloader = (function () {
 
     FlavorDownloader.prototype.stop = function () {
         var that = this;
-        if (that.stopPromise)
-        {
-            // Stop already requested
-            that.logger.info('stop requested - ignoring request as this flavor downloader either already stopped or is currently stopping');
-            return that.stopPromise;
-        }
-
         var d = Q.defer();
         that.stopPromise = d.promise;
         that.stopPromise.then(function(){
             that.runStatus = 'stopped';
-            that.emit("stopped", that.flavor);
+            that.emit('stopped', that.flavor);
+        });
+        that.logger.info("Stopping flavor downloader");
+        that.shouldRun = false;
+        downloaderEnd.promise.then(function() {
+            that.logger.info("FlavorDownloader Stopped");
+            delete FlavorDownloader.activeInstances[that.id];
+            d.resolve();
         });
 
-        var teardown = Q();
-        if (that.runStatus === 'running')
-        {
-            var teardownInterval = config.get("flavorDownloaderTeardownInterval");
-            that.logger.info('stop requested - allowing ' + teardownInterval + 'ms for orderly tear-down');
-            teardown = Q.delay(teardownInterval).then(function(){
-                that.logger.info('stopping flavor downloader after allowing orderly tear-down');
-            });
-        }
+        return d.promise;
```

Comment:
FlavorDownloader.prototype.stop = function () {
        var that = this;
        that.logger.info("Stopping flavor downloader");
        that.shouldRun = false;
        return downloaderEnd.promise.then(function() {
            that.logger.info("FlavorDownloader Stopped");
            delete FlavorDownloader.activeInstances[that.id];
            that.runStatus = 'stopped';
            that.emit('stopped', that.flavor);
            
        });


    };